changes/process_changes.py

```diff
@@ -0,0 +1,124 @@
+# coding: utf-8
+
+from __future__ import absolute_import, division, print_function
+
+import argparse as A
+from funcy import memoize
+from glob import glob
+import os
+from os import path as P
+import re
+import yaml
+
+
+TN_RE = re.compile(r"[A-Z]\d\d\d-\d\d\d")
+
+
+def all_entries():
+    return [f for f in glob('*.yaml') if not f == 'entry_schema.yaml']
+
+
+types_to_header = {
+    'api-change': 'breaking changes',
+    'new-feature': 'new features'
+}
+
+
+def create_rst(change_type="api-change"):
+    """
+    Create an RST file from all the change entries for a given ``change_type``.
+    """
+    header_chunk = types_to_header[change_type]
+
+    def header(title, header_char):
+        """
+        Format a header
+        """
+        return "{}\n{}\n".format(title, header_char * len(title))
+
+    def field(name, value):
+        """
+        Format a field list entry
+        """
+        return ":{}:\n    {}".format(name, value)
+
+    def format_apis(apis):
+        """
+        Format the API field content. If absent, return "all".
+        """
+        if apis:
+            return ", ".join(apis)
+        else:
+            return "all"
+
+    entries = []
+
+    for fname in all_entries():
+        tn = fname.split(".")[0]
+        with open(fname) as f:
+            entry = yaml.safe_load(f)
+            entry['tn'] = tn
+            validate_entry(tn, entry)
+            if entry['type'] == change_type:
+                entries.append(entry)
+
+    entries = list(sorted(entries, key=lambda e: e['date'], reverse=True))
+
+    print(header("Libadalang API {}".format(header_chunk), "#"))
+
+    for entry in entries:
+        print(header(entry["title"], "="))
+        print(entry["description"])
+        print(field("tn", entry["tn"]))
+        print(field("date", entry["date"]))
```

Comment:
As discussed live, the date has no specific meaning (it’s the the date of merge in `stable` branches) and used only for sorting entries. So I think it should be left out of the output to avoid confusion with meaningful dates (date of merge in `master`/`stable`).

Reply:
Hmm I wonder if I agree. In the current situation the date will be approximately the date of the merge in master, or at least will give a rough idea. This can still be useful to people, as it gives a lower bound about in which version of LAL you can expect the feature to be here. Otherwise, this info will be pretty hard to find.

Reply:
I’m only aware of AdaCore engineers who want to have this info, and they care only about the `stable` branch, for which we agreed to have a specific workflow (record the date of the branch bump during the bump IIRC), so I’m not convinced this will be useful in practice. My preference goes towards starting to be conservative and not show some approximate info.

Reply:
I think you are wrong: AdaCore engineers care about the master branch, specifically about the diff of changes between current master and current stable, since usually they care about this list when we are going to bump stable. Could you clarify your reasoning or tell me if you missed something here ?

Reply:
I forgot about this use case, thanks.

Now, as I’ve written in private, we need to have data about which change appears in which release. We could use the same mechanism to track which change is in `stable` and which isn’t in any release (not even wavefronts). That will provide the info people want without showing approximate dates. Thoughts?

Reply:
I'll remove the date for now. Closing this and let's discuss this via a more synchronous medium :)